include/seqan3/range/decorator/gap_decorator.hpp

```diff
@@ -576,12 +179,16 @@ ame Arithmetic operators
      * Average and worst case (insertion before last gap): \f$O(k)\f$,
      * Best case (back insertion): \f$O(\log k)\f$.
      */
-    iterator insert_gap(iterator const it, size_type const count = 1)
+    template <typename iterator_t>
+    iterator_t insert_gap(iterator_t const it, size_type const count = 1)
     {
+        static_assert(std::same_as<iterator_t, const_iterator> || std::same_as<iterator_t, const_iterator_ra>,
+                      "Expects class iterator or iterator_ra type.");
+
         if (!count) // [[unlikely]]
             return it;
 
-        size_type const pos = std::distance(begin(), it);
+        size_type const pos = it - begin();
```

Comment:
Wouldn't this require random access iterators?
Why not just leave the `std::distance`

Reply:
Begin resolves to the concept and the gap decorator only provides in the default way a "bidirectional iterator" also it supports all interfaces of a random access iterator. So internally I can use the faster implementation because I know the type. Calling distance would otherwise always fallback to linear runtime.

Reply:
Ah ok, makes sense 👍